classes/admin_setting_check.php

```diff
@@ -0,0 +1,83 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace search_elastic;
+
+use admin_setting;
+use core\check\check;
+
+/**
+ * Admin setting for check api.
+ *
+ * @package     search_elastic
+ * @copyright   Matthew Hilton <matthew.hilton@catalyst-au.net>
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class admin_setting_check extends admin_setting {
+    /** @var check $check The check to display */
+    private $check;
+
+    /**
+     * Creates check setting.
+     *
+     * @param string $name name of setting
+     * @param string $heading title of setting
+     * @param check $check check to display
+     */
+    public function __construct(string $name, string $heading, check $check) {
+        $this->nosave = true;
+        $this->check = $check;
+        parent::__construct($name, $heading, '', '');
+    }
+
+    /**
+     * Returns setting (unused)
+     *
+     * @return true
+     */
+    public function get_setting() {
+        return true;
+    }
+
+    /**
+     * Writes the setting (unused)
+     *
+     * @param mixed $data
+     */
+    public function write_setting($data) {
+        return '';
+    }
+
+    /**
+     * Ouputs the admin setting HTML to be rendered.
+     *
+     * @param mixed $data
+     * @param string $query
+     * @return string html
+     */
+    public function output_html($data, $query = '') {
+        global $OUTPUT;
+
+        // Run the check and get the result.
+        $checkresult = $this->check->get_result();
+
+        $resulthtml = $OUTPUT->check_result($checkresult);
```

Comment:
Can you please make this a notification, success / warn / error etc

Reply:
With 76b9243ca45219545437a089ce50fe47ca6c49e9 I have updated the formatting to use the notification. Note I only included error and ok notifications, since this check only uses n/a, error and ok check results.

In the core version it will support the rest e.g. warning, info, etc..

![image](https://github.com/catalyst/moodle-search_elastic/assets/17095477/c269d36d-9de0-4332-95fb-ba4024ecfbd1)
![image](https://github.com/catalyst/moodle-search_elastic/assets/17095477/a240a3dc-6664-45f5-b2a3-550634ebedec)